src/main/java/com/battleship/backend/models/Board.java

```diff
@@ -79,6 +79,24 @@ public boolean isPermutationOfShipList(ArrayList<Ship> otherShipList){
         return count == shipList.size() && count == otherShipList.size();
     }
 
+    public boolean isEveryShipSunk(){
+        for(Ship ship : shipList){
+            if(!ship.checkIsSunk()){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Ship getShip(String shipName){
+        for(Ship ship : shipList){
+            if(ship.name.equals(shipName)){
+                return ship;
+            }
+        }
+        return null;
```

Comment:
Would this function ever get called with a `shipName` that doesn't exist in the `shipList`? If not, is the `return null` necessary?

Reply:
The return null was necessary to get the code to compile. Even though we will only call it with ship names that exist, if the conditional isn't met then the code would break.